src/main/java/org/ingress/cartms/client/decoder/CustomErrorDecoder.java

```diff
@@ -0,0 +1,37 @@
+package org.ingress.cartms.client.decoder;
+
+import static org.ingress.cartms.client.decoder.JsonNodeFieldName.MESSAGE;
+import static org.ingress.cartms.exception.ExceptionConstraints.CLIENT_ERROR;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import feign.Response;
+import feign.codec.ErrorDecoder;
+import org.ingress.cartms.exception.CustomFeignException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class CustomErrorDecoder implements ErrorDecoder {
+
+    private static final Logger log = LoggerFactory.getLogger(CustomErrorDecoder.class);
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        var errorMessage = CLIENT_ERROR;
```

Comment:
[Optional]: you can initialize `var statusCode = response.status()` and use the following code snippet (2 used case)